packages/api/core/src/api/publish.ts

```diff
@@ -55,137 +64,227 @@ export interface PublishOptions {
+        persistentOutput: true,
+      },
+    },
+  ];
 
-  const resolvedDir = await resolveDir(dir);
-  if (!resolvedDir) {
-    throw new Error('Failed to locate publishable Electron application');
-  }
-  dir = resolvedDir;
+  const runner = new Listr<PublishContext>(
+    [
+      {
+        title: 'Loading configuration',
+        task: async (ctx) => {
+          const resolvedDir = await resolveDir(providedDir);
+          if (!resolvedDir) {
+            throw new Error('Failed to locate publishable Electron application');
+          }
 
-  // const testPlatform = makeOptions.platform || process.platform as ForgePlatform;
-  if (!publishTargets) {
-    publishTargets = forgeConfig.publishers || [];
-    // .filter(publisher => (typeof publisher !== 'string' && publisher.platforms)
-    //   ? publisher.platforms.includes(testPlatform) : true);
-  }
-  publishTargets = (publishTargets as ForgeConfigPublisher[]).map((target) => {
-    if (typeof target === 'string') {
-      return (
-        (forgeConfig.publishers || []).find((p: ForgeConfigPublisher) => {
-          if (typeof p === 'string') return false;
-          if ((p as IForgePublisher).__isElectronForgePublisher) return false;
-          return (p as IForgeResolvablePublisher).name === target;
-        }) || { name: target }
-      );
-    }
-    return target;
-  });
-
-  for (const publishTarget of publishTargets) {
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    let publisher: PublisherBase<any>;
-    if ((publishTarget as IForgePublisher).__isElectronForgePublisher) {
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      publisher = publishTarget as PublisherBase<any>;
-    } else {
-      const resolvablePublishTarget = publishTarget as IForgeResolvablePublisher;
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      let PublisherClass: any;
-      await asyncOra(`Resolving publish target: ${chalk.cyan(resolvablePublishTarget.name)}`, async () => {
-        PublisherClass = requireSearch(dir, [resolvablePublishTarget.name]);
-        if (!PublisherClass) {
-          throw new Error(
-            `Could not find a publish target with the name: ${resolvablePublishTarget.name}. Make sure it's listed in the devDependencies of your package.json`
+          ctx.dir = resolvedDir;
+          ctx.forgeConfig = await getForgeConfig(resolvedDir);
+        },
+      },
+      {
+        title: 'Resolving publish targets',
+        task: async (ctx: PublishContext, task: ForgeListrTask<PublishContext>) => {
+          const { dir, forgeConfig } = ctx;
+
+          // const testPlatform = makeOptions.platform || process.platform as ForgePlatform;
+          if (!publishTargets) {
+            publishTargets = forgeConfig.publishers || [];
+            // .filter(publisher => (typeof publisher !== 'string' && publisher.platforms)
+            //   ? publisher.platforms.includes(testPlatform) : true);
```

Comment:
Can we just delete this comment?

Reply:
I'm currently staring at this comment trying to understand why it was removed (this code was copied over from the old impl)